presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetBinaryColumnReader.java

```diff
@@ -54,7 +54,7 @@ protected void readValue(BlockBuilder blockBuilder, Type type)
             }
             type.writeSlice(blockBuilder, value);
         }
-        else {
+        else if (isNullValue()) {
             blockBuilder.appendNull();
         }
     }
```

Comment:
@nezihyigitbasi This is a valid case. Value can be undefined when it belongs to block under some nested structure. 
For primitive columns it's true, that their value either defined or null.
For columns with nested structure it's different, because null can be at any level of optional field.

For example, column can be a struct with two nested optional fields A.B
In this case:
A can be defined or null
B can be defined, null or not defined (meaning A was null). 

Presto spi blocks contain only defined and null values.  So when block with values for B field is being written, cases where A field is null need to skipped. 

When complex type is being decoded, be it RowType.getObjectValue(), ArrayType.getObjectValue() or MapType.getObjectValue() they all first check parent value for being not null and only if it's not, decode nested values from underlying blocks.
This means, that value from block B will be read only if corresponding parent field A value was not null.